Oleksandr.Svidovyi/Task_6/src/main/java/com/hillel/task_6/service/ClientService.java

```diff
@@ -0,0 +1,64 @@
+package com.hillel.task_6.service;
+
+import com.hillel.task_6.model.Account;
+import com.hillel.task_6.model.Client;
+import com.hillel.task_6.repository.ClientRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+@Service
+public class ClientService {
+    @Autowired
+    private ClientRepository clientRepository;
+
+
+    public Client getClient(int id) {
+        return clientRepository.getClientById(id);
+    }
+
+
+    public void saveClient(Client client) {
+        clientRepository.save(client);
+    }
+
+
+    public void removeClient(int id) {
+        clientRepository.deleteClientById(id);
+    }
+
+
+    public List<Client> getClientsList() {
+        return clientRepository.getClientsList();
+    }
+
+
+    public void saveAccount(int clientId, Account account) {
+        Client client = clientRepository.getClientById(clientId);
+
+        if (client.getAccounts() == null) {
+            client.setAccounts(new ArrayList<Account>());
+            client.getAccounts().add(account);
+        } else client.getAccounts().add(account);
+    }
+
+
+    public void removeAccount(int id, Account account) {
+        clientRepository.getClientById(id).getAccounts().remove(account);
```

Comment:
What if clientService.getClientById(id) is null?

add this check

Reply:
Since we fetch clientId from URL wich is formed from concrete client, I asume client already exists.

Method clientService.getAccountsByClientId(int clientId) was added.

Reply:
Since we fetch clientId from a concrete client in clientsList.jsp, I asume client already exists. And in ApiAccountController.deleteAccount() method we have this check in line 60.